src/herder/test/HerderTests.cpp

```diff
@@ -2164,6 +2166,45 @@ TEST_CASE("SCP State", "[herder][acceptance]")
     LedgerHeaderHistoryEntry lcl;
     uint32_t numLedgers = 5;
     uint32_t expectedLedger = LedgerManager::GENESIS_LEDGER_SEQ + numLedgers;
+    std::unordered_set<Hash> knownTxSetHashes;
+
+    auto checkTxSetHashesPersisted =
+        [&](Application::pointer app,
+            std::optional<
+                std::unordered_map<uint32_t, std::vector<SCPEnvelope>>>
+                expectedSCPState) {
+            // Check that node0 restored state correctly
+            auto& herder = static_cast<HerderImpl&>(app->getHerder());
+            auto limit = app->getHerder().getMinLedgerSeqToRemember();
+
+            std::unordered_set<Hash> hashes;
+            for (auto i = app->getHerder().trackingConsensusLedgerIndex();
+                 i >= limit; --i)
+            {
+                if (i == LedgerManager::GENESIS_LEDGER_SEQ)
+                {
+                    continue;
+                }
+                auto msgs = herder.getSCP().getLatestMessagesSend(i);
+                if (expectedSCPState.has_value())
```

Comment:
nit: `if (expectedSCPState)` is sufficient, though I suppose you could use the explicit check for expressivity, so up to you

Reply:
kept it as-is if you don't mind: has_value is used pretty commonly in the codebase, and I do like the explicitness of it